### PR TITLE
Init go implementation

### PR DESCRIPTION
**Pull request by mskrip**

### CODE REVIEW

File: objects.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package main
+
+import "github.com/faiface/pixel"
+
+// Gravity is vector representing standard gravity force
+var Gravity = pixel.Vec{
+	X: 0.0,
```
Comment (mitas1) [inline objects.go]: Why this is uppercased?

File: objects.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	ExplicitEuler PositionIntegrationMethod = iota
+	// MidPoint is a position integration method based on Mid Point algorithm
+	MidPoint PositionIntegrationMethod = iota
+	// Verlet is a position integration method based on Verlet's algorithm
+	Verlet PositionIntegrationMethod = iota
+)
+
+// Particle represents particle object
+type Particle struct {
+	pos      pixel.Vec
```
Comment (mitas1) [inline objects.go]: IMO, `position` would be more suitable.

File: physical-based-animations.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/faiface/pixel/pixelgl"
+)
+
+func updateParticles(particles []Particle, batch *pixel.Batch, dt float64, cam pixel.Matrix) {
+	for i := 0; i < len(particles); i++ {
+		newPos, err := newPosition(particles[i], dt, ExplicitEuler)
+		if err != nil {
+			fmt.Println(err.Error())
+		}
+		particles[i].pos = newPos
```
Comment (mitas1) [inline physical-based-animations.go]: Missing new line.

File: physical-based-animations.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	case ExplicitEuler:
+		return particle.ExplicitEulerIntegrator(dt), nil
+	case MidPoint:
+		return particle.pos, errors.New("Unimplemented")
+	case Verlet:
+		return particle.pos, errors.New("Unimplemented")
+	default:
+		return particle.pos, errors.New("Unknown method of position integration")
+	}
+
```
Comment (mitas1) [inline physical-based-animations.go]: Unnecessary new line.

File: utils.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+	"github.com/faiface/pixel"
+)
+
+func loadPicture(path string) (pixel.Picture, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
```
Comment (mitas1) [inline utils.go]: missing new line

File: utils.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	img, _, err := image.Decode(file)
+	if err != nil {
+		return nil, err
+	}
+	return pixel.PictureDataFromImage(img), nil
```
Comment (mitas1) [inline utils.go]: new line

File: objects.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+			particle.pos.Y >= minY {
+			aliveParticles = append(aliveParticles, particle)
+		}
+	}
+	particleSystem.particles = append([]Particle{}, aliveParticles...)
+}
+
+// ParticleSystem represents system of particles with and rate of particle generation per second
+type ParticleSystem struct {
+	pos       pixel.Vec
```
Comment (mitas1) [inline objects.go]: As I mentioned above, `position` would be more suitable . What do you think?

File: objects.go [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	pps       int
```
Comment (mitas1) [inline objects.go]: What `pps` means? Particle per second?

File: physical-based-animations.go [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+		frames++
+		select {
+		case <-second:
+			particleSystem.killOldParticles(
+				win.Bounds().Min.X,
+				win.Bounds().Max.X,
+				win.Bounds().Min.X,
+			)
+			win.SetTitle(fmt.Sprintf("%s | FPS: %d | particles %d", cfg.Title, frames, len(particleSystem.particles)))
```
Comment (mitas1) [inline physical-based-animations.go]: Please, split this line after 100 characters.

File: physical-based-animations.go [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+			frames = 0
+		default:
```
Comment (mitas1) [inline physical-based-animations.go]: Is this `default` really necessary?

File: objects.go (hunk 8 above)
Comment (mskrip) [reply objects.go]: Yes, any suggestions for better name?

File: physical-based-animations.go (hunk 10 above)
Comment (mskrip) [reply physical-based-animations.go]: It is, without it, the `select` statement would wait for every `second` to tick

File: objects.go (hunk 1 above)
Comment (mskrip) [reply objects.go]: Because it is an exported variable and those should be uppercased in `Go`

**Pull request merged**